BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/TelemetryExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using Microsoft.ApplicationInsights.Channel;
+
+    /// <summary>
+    /// Extensions for ITelemetry interface.
+    /// </summary>
+    public static class TelemetryExtensions
+    {
+        /// <summary>
+        /// Sets envelope name for ITelemetry object.
+        /// </summary>
+        /// <param name="telemetry">ITelemetry object to set envelope name for.</param>
+        /// <param name="envelopeName">Envelope name to use for ITelemetry object.</param>
+        /// <exception cref="ArgumentException">Concrete implementation of ITelemetry object does not expose envelope name.</exception>
+        public static void SetEnvelopeName(this ITelemetry telemetry, string envelopeName)
+        {
+            if (telemetry is IAiSerializableTelemetry aiSerializableTelemetry)
+            {
+                aiSerializableTelemetry.TelemetryName = envelopeName;
+            }
+            else
+            {
+                throw new ArgumentException("Provided telemetry object does not support envelope name.", nameof(telemetry));
```

Comment:
Not a big fan of this throw, but currently only `IAiSerializableTelemetry` exposes envelope name and it's an internal-only interface that was meant to remain internal. Currently, any out-of-the-box `ITelemetry` implements `IAiSerializableTelemetry`, but that may not be the case if someone implements custom `ITelemetry` or we later ship non-serializable type. Thoughts?

Reply:
Just use default `"Event"`

Reply:
I see. I'd suggest using the default name here as well.

Reply:
really don't want exceptions in serialization